src/utils/nameLookup.ts

```diff
@@ -0,0 +1,44 @@
+import { AddressLookupArgs, AddressLookupResult, DomainLookupArgs, DomainLookupResult } from '@metamask/snaps-sdk';
+
+import { CHAIN_CODES_AND_TOKEN_CODES_BY_NETWORK, PROTOCOL_NAME } from '../constants';
+import { validateFioHandle } from './validation/fio';
+import { getPublicAddressByFioHandle } from './chain/chain-get-info';
+
+export const nameLookup = async (request: AddressLookupArgs | DomainLookupArgs): Promise<AddressLookupResult | DomainLookupResult | null> => {
+  const { chainId, domain } = request;
+
+  if (domain && chainId) {
+    const isDomainValid = validateFioHandle(domain);
+    const { chainCode, tokenCode } = CHAIN_CODES_AND_TOKEN_CODES_BY_NETWORK[chainId] || {};
+
+    if (!chainCode || !tokenCode || !isDomainValid) {
+      return null;
+    }
+
+    try {
+      const publicAddress = await getPublicAddressByFioHandle({
+        apiUrl: 'https://fio.blockpane.com',
```

Comment:
Should be https://chain.fio.net
Also should it not be moved to constants.ts?

Reply:
That was test version. Updated to https://chain.fio.net